syntax = "proto3";

package ecommerce;

import "google/protobuf/empty.proto";

service ProductInfo {
    rpc addProduct(Product) returns (ProductID);
    rpc getProduct(ProductID) returns (Product);
    rpc deleteProduct(ProductID) returns (google.protobuf.Empty);
}

message Product {
    string id = 1;
    string name = 2;
    string description = 3;
    float price = 4;
}

message ProductID {
    string value = 1;
}
